reana_commons/k8s/kerberos.py

```diff
@@ -115,14 +115,22 @@ def get_kerberos_k8s_config(
                 f"while ! test -f {KRB5_STATUS_FILE_LOCATION}; do "
                 f"if [ $SECONDS -ge {KRB5_TICKET_RENEW_INTERVAL} ]; then "
                 'echo "Renewing Kerberos ticket: $(date)"; kinit -R; SECONDS=0; fi; '
-                f"sleep {KRB5_STATUS_FILE_CHECK_INTERVAL}; done"
+                # wait until status file is created or for a given timeout, whichever comes first
+                f"inotifywait --quiet --format 'Created file %w%f, stopping inotifywait' --timeout {KRB5_STATUS_FILE_CHECK_INTERVAL} --event create {KRB5_TOKEN_CACHE_LOCATION}; "
+                "done"
             )
         ],
         "name": KRB5_RENEW_CONTAINER_NAME,
         "imagePullPolicy": "IfNotPresent",
         "volumeMounts": [secrets_volume_mount] + volume_mounts,
         "env": env,
         "securityContext": {"runAsUser": kubernetes_uid},
+        "lifecycle": {
+            # make sure we stop the sidecar container when the pod is stopped,
+            # for example when the run-batch pod is terminated by reana-workflow-controller
+            # after the workflow finishes (either successfully or with an error)
+            "preStop": {"exec": {"command": ["touch", KRB5_STATUS_FILE_LOCATION]}}
```

Comment:
Works nicely, the sidecar is stopped "immediately" :+1:  

Two comments:

- For the commit message headline, instead of saying "immediately", perhaps we could say "as soon as possible" (gently hinting at the fact that the main job must finish first).

- Does this change mean that we could in theory increase the value of `KRB5_STATUS_FILE_CHECK_INTERVAL` from the current values of 15 seconds?  (Not that it would be necessary...)